src/main/java/com/epam/esm/integration/sqlrepo/MySQLRepository.java

```diff
@@ -0,0 +1,243 @@
+                    }
+                } else giftCertificate.getTagsList().add(new Tag(rs.getLong("tags.id"), rs.getString("tags.name")));
+            }
+            return giftCertificate == null ? Optional.empty() : Optional.of(giftCertificate);
+        }, id);
+    }
+
+    @Override
+    public Optional<List<GiftCertificate>> getAllGiftCertificates() {
+        String SELECT_FROM_GiftCertificates = SQLQuery.SELECT_FROM_CERTIFICATES;
+        return Optional.of(jdbcTemplate.query(SELECT_FROM_GiftCertificates, (ResultSet resultset, int i) ->
+                new GiftCertificate(resultset.getLong("id"),
+                        resultset.getString("name"),
+                        resultset.getString("description"),
+                        resultset.getBigDecimal("price"),
+                        Duration.ofDays(resultset.getLong("duration")).toString(),
+                        resultset.getTimestamp("create_date").toLocalDateTime().toString(),
+                        resultset.getTimestamp("last_update_date").toLocalDateTime().toString(),
+                        Collections.singletonList(null))));
+    }
+
+    @Override
+    public Optional<GiftCertificate> createNewGiftCertificate(GiftCertificate giftCertificate) {
+       return transactionTemplate.execute(status -> {
+            try {
+                createCertificateInTransaction(giftCertificate, keyHolder);
+                long giftCertificate_id = (Objects.requireNonNull(keyHolder.getKey())).longValue();
+                if (giftCertificate.getTagsList() != null && !giftCertificate.getTagsList().isEmpty()) {
+                    addEachNewTagToDbAndCreateRelationships(giftCertificate_id, giftCertificate, keyHolder);
+                }
+                return getGiftCertificateById(giftCertificate_id);
+            } catch (Exception e) {
+                status.setRollbackOnly();
+                throw new CertificateTransactionException("Exception during create certificate. Certificate has not been created.");
+            }
+        });
+    }
+
+
+    @Override
+    public boolean deleteGiftCertificateById(Long id) {
+        return jdbcTemplate.update(SQLQuery.DELETE_FROM_CERTIFICATES_WHERE_ID, id) == 1;
+    }
+
+    @Override
+    public Optional<GiftCertificate> updateGiftCertificateById(Long id, GiftCertificate giftCertificate) {
+        return transactionTemplate.execute(status -> {
+            try {
+                jdbcTemplate.update(con -> {
+                    PreparedStatement ps = con.prepareStatement(SQLQuery.UPDATE_CERTIFICATES_ALL_FIELDS);
+                    ps.setString(1, giftCertificate.getName());
+                    ps.setString(2, giftCertificate.getDescription());
+                    ps.setBigDecimal(3, giftCertificate.getPrice());
+                    ps.setInt(4, Integer.parseInt(giftCertificate.getDuration()));
+                    ps.setTimestamp(5, Timestamp.valueOf(LocalDateTime.now()));
+                    ps.setLong(6, id);
```

Comment:
Those are magic number, either create a dedicated constatn for each
int GIFT_CERTIFICATE_NAME_PARAMETER_INDER = 1;

or use inside method incrementing approach
int paramNumber = 1;
and then
ps.setString(i++, giftCertificate.getName());
ps.setString(i++, giftCertificate.getDescription());
ps.setBigDecimal(i++, giftCertificate.getPrice());